invisible_cities/cities/psf_liquidata.py

```diff
@@ -0,0 +1,193 @@
+from functools import partial
+
+import numpy  as np
+import tables as tb
+import pandas as pd
+
+from invisible_cities.reco.psf_functions import create_psf
+from invisible_cities.reco.psf_functions import hdst_psf_processing
+
+from liquidata import flat
+from liquidata import get
+from liquidata import put
+from liquidata import pipe
+from liquidata import out
+
+from invisible_cities.io  .kdst_io        import psf_writer
+from invisible_cities.core.core_functions import in_range
+from invisible_cities.core                import system_of_units as units
+from invisible_cities.cities.components   import city
+
+from invisible_cities.database.load_db import DataSiPM
+
+from typing import Sequence
+
+
+# TODO: @Aretno to decide the name of this city
+# TODO: Add city description and etymology
+@city
+def create_psf_city( files_in, file_out, compression, event_range
+                   , detector_db, run_number
+                   , xrange      : Sequence[float]
+                   , yrange      : Sequence[float]
+                   , zbins       : Sequence[float]
+                   , xsectors    : Sequence[float]
+                   , ysectors    : Sequence[float]
+                   , bin_size_xy : float           = 1 * units.mm
+                   ):
+    sipms   = DataSiPM(detector_db, run_number)
+    ranges  = xrange  , yrange
+    sectors = xsectors, ysectors
+
+    nbin      = (np.diff(ranges, axis=1) / bin_size_xy).astype(int).flatten()
+    bin_edges = [np.linspace(*limits, n+1) for limits, n in zip(ranges, nbin)]
+
+    columns_to_drop     = "Xrms Yrms Qc Ec".split()
+    drop_unused_columns = get.dst    *      partial(column_dropper, columns_to_drop)   >> put.dst
+    create_slices_z     = get.dst    * flat(partial(split_in_z, zbins, ranges, sipms)) >> put.z.dst_z
+    create_sectors_xy   = get.dst_z  * flat(partial(split_in_xy, sectors, bin_edges))  >> put.x.y.dst_xy
+    compute_psf         = get.dst_xy *      partial(psf_builder, bin_edges)            >> put.factors.entries.centers
```

Comment:
`liquidata`s `*` (and `>>`) could be very confusing, especially if liquidata code is interleaved with non-liquidata code using these operators in their 'normal' sense. *Perhaps* we should consider a style where the `liquidata` names (especially `get` and `put`) are imported uppercased, to give a visual warning that something unusual is happening.

You can see this style being used [here](https://jacg.github.io/liquidata/#still-with-me).

We might even add these names as synonyms inside `liquidata` itself.

Reply:
I am ok with both options. I am already familiar with `get` and `put`, perhaps we should get the view from someone less involved with this code.

Reply:
Yes, I'd love to get some input from someone less involved with the code. I'd love to get any indication that we are not alone in this universe.

Reply:
Not sure if i'm fully eligible as 'less involved' as I already know what the code should do (but have no idea about `liquidata`) but this seems pretty straightforward and readable. I get the confusing part, though, so it could be useful to have the uppercase hint (although I guess that in a liquidata world one could infere the use of the characters once you have been warned with the `@city` decorator).

Reply:
I don't think that `@city` is enough warning, because within city code one can find uses of `*` which mean multiplication, so inside city code we might end up having such lines interspersed with lines on which `*` means `liquidata` argument selection.

It needs to be clear to the reader that we're in a `liquidata` context rather than an arithmetic (or other normal) context.

But I have no strong opinions on whether uppercasing `liquidata` names helps.